python/astro_metadata_translator/translator.py

```diff
@@ -501,6 +502,69 @@ def validate_value(value, default, minimum=None, maximum=None):
                 value = default
         return value
 
+    @staticmethod
+    def is_keyword_defined(header, keyword):
+        """Return `True` if the value associated with the named keyword is
+        present in the supplied header and defined.
+
+        Parameters
+        ----------
+        header : `dict`-lik
+            Header to use as reference.
+        keyword : `str`
+            Keyword to check against header.
+
+        Returns
+        -------
+        is_defined : `bool`
+            `True` if the header is present and not-`None`. `False` otherwise.
+        """
+        if keyword not in header:
+            return False
+
+        if header[keyword] is None:
+            return False
+
+        # Special case Astropy undefined value
+        if isinstance(header[keyword], astropy.io.fits.card.Undefined):
+            return False
+
+        return True
+
+    def is_key_ok(self, keyword):
```

Comment:
I imagine there is some clever reason that explains why this is necessary/different to `is_keyword_defined()`, but as it stands, both the docstrings, and the fact that this just directly calls `is_keyword_defined()` seems odd to me. Is this just a pass-through with a more descriptive method name to allow for neater extension in the future, should it be needed, or something?

Reply:
There have to be two different methods because I need the `can_translate` methods to be able to do the same definedness checks. I'm happy to change the names. `is_keyword_defined` is a static method that can be used by class methods. `is_key_ok` is for instances of translations.

Reply:
No no, it's fine, that makes sense, thanks!